src/Hng.Web/Controllers/ProductController.cs

```diff
@@ -73,5 +72,39 @@ public async Task<ActionResult<IEnumerable<CategoryDto>>> GeProductCategories()
                 categories
             });
         }
+
+        /// <summary>
+        /// Edit user products with update timestamp
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="updateProductDto"></param>
+        /// <returns></returns>
+        [HttpPut("{id:guid}")]
+        [Authorize]
+        [ProducesResponseType(typeof(ProductDto), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<IActionResult> UpdateProduct(Guid id, [FromBody] UpdateProductDto updateProductDto)
+        {
+            var validationResult = new UpdateProductDtoValidator().Validate(updateProductDto);
+            if (!validationResult.IsValid)
```

Comment:
1. This controller is code-heavy. Validations do not happen in the controller. 

1. Delegate validation logic to the command handlers, to keep the controller's responsibilities focused on handling HTTP requests and responses.

Reply:
have make neccessary changes @Godhanded @chimaihueze